tobac/testing.py

```diff
@@ -405,81 +509,92 @@ def make_feature_blob(in_arr, h1_loc, h2_loc, v_loc = None,
         start_loc = 1
         v_min = 0
         v_max = in_arr.shape[start_loc]
-        start_v = round(max(v_min, v_loc - v_size/2))
-        end_v = round(min(v_max-1, v_loc + v_size/2))
+        start_v = int(np.ceil(max(v_min, v_loc - v_size / 2)))
+        end_v = int(np.ceil(min(v_max - 1, v_loc + v_size / 2)))
         if v_size > v_max - v_min:
             raise ValueError("v_size larger than domain size")
 
-
     # Get min/max coordinates for hdim_1 and hdim_2
     # Min is inclusive, end is exclusive
     h1_min = 0
     h1_max = in_arr.shape[start_loc]
 
     h2_min = 0
-    h2_max = in_arr.shape[start_loc+1]
+    h2_max = in_arr.shape[start_loc + 1]
 
-    if ((h1_size > h1_max - h1_min) or (h2_size > h2_max - h2_min)):
+    if (h1_size > h1_max - h1_min) or (h2_size > h2_max - h2_min):
         raise ValueError("Horizontal size larger than domain size")
 
     # let's get start/end x/y/z
-    start_h1 = round(h1_loc - h1_size/2)
-    end_h1 = round(h1_loc + h1_size/2)
-
-    start_h2 = round(h2_loc - h2_size/2)
-    end_h2 = round(h2_loc + h2_size/2)
-    in_arr = set_arr_2D_3D(in_arr, amplitude, start_h1, end_h1, 
-                           start_h2, end_h2, start_v, end_v)
-    return in_arr    
-
-            
-                
-        
-def set_arr_2D_3D(in_arr, value, start_h1, end_h1, start_h2, end_h2,
-                  start_v = None, end_v = None):
-    '''Function to set part of `in_arr` for either 2D or 3D points to `value`.
-    If `start_v` and `end_v` are not none, we assume that the array is 3D. If they 
-    are none, we will set the array as if it is a 2D array. 
+    start_h1 = int(np.ceil(h1_loc - h1_size / 2))
+    end_h1 = int(np.ceil(h1_loc + h1_size / 2))
+
+    start_h2 = int(np.ceil(h2_loc - h2_size / 2))
+    end_h2 = int(np.ceil(h2_loc + h2_size / 2))
```

Comment:
Happy with these changes, this should also make the sizes more consistent than those given by using round